src/main/scala/com/gu/contentapi/services/PageLookup.scala

```diff
@@ -0,0 +1,57 @@
+package com.gu.contentapi.services
+
+import com.gu.contentapi.client._
+import com.gu.contentapi.client.model.v1.SearchResponse
+import com.gu.contentapi.client.model.SearchQuery
+import com.typesafe.scalalogging.StrictLogging
+import scala.concurrent.duration._
+import scala.concurrent.{ Await, Future }
+import com.gu.contentapi.Config.capiKey
+
+object PageLookup extends StrictLogging {
+
+  import scala.concurrent.ExecutionContext.Implicits.global
+
+  sealed trait ResponseFromCapiQuery
+  case class SuccessfulQuery(searchResponse: SearchResponse) extends ResponseFromCapiQuery
+  case class FailedQuery(errorMsg: String) extends ResponseFromCapiQuery
+
+  val client = new GuardianContentClient(capiKey)
+
+  var cache = Map[String, String]()
```

Comment:
It's good practice to use an AtomicReference when you have some mutable state in your application that you want to do atomic operations on (although it's fine in your case as there's only a single thread).

https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/atomic/AtomicReference.html

See for example: https://github.com/guardian/facebook-ia-rss/blob/3c2d12126f99fd87a40fcab6c90c233f29318055/app/com/gu/facebook/PublicationHealth.scala